tools/ci/prepare-pull-credentials.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e
+
+if [[ "$DOCKERHUB_PULL_CREDENTIAL" == "" ]]; then
+    >&2 echo "No docker pull credential information specified"
+else
+    DOCKER_USER=$(echo "$DOCKERHUB_PULL_CREDENTIAL" | cut -d ':' -f 1)
+    DOCKER_PWD=$(echo "$DOCKERHUB_PULL_CREDENTIAL" | cut -d ':' -f 2)
+    echo -n "$DOCKER_PWD" | docker login -u "$DOCKER_USER" --password-stdin > /dev/null
+    cat <<EOF > "$HOME/.docker/k3d-registry.yaml"
```

Comment:
Generate a registry file to be used by `k3d`
docs: https://k3d.io/v5.2.0/usage/registries/#authenticated-registries

Reply:
If you feel the need to add a comment to a PR to explain your code then it should probably just be a comment in the code in the first place